src/NBitcoin/BlockStake.cs

```diff
@@ -195,6 +195,12 @@ public class PosBlockHeader : BlockHeader
 
         /// <inheritdoc />
         public override uint256 GetHash()
+        {
+            return this.GetHash(null);
+        }
+
+        /// <inheritdoc />
+        public override uint256 GetHash(ConsensusFactory consensusFactory)
```

Comment:
I think we should wait with this change for now its not necessarily for the Network changes.
Stratis header is the same as bitcoin so it will work fine and I would try to avoid passing the factory in the method.

This is probably a deeper changes that needs more thinking.
An idea is to pass the ConsensusFacotyr to the Block/BlockHeader/Transaction constructors (as they are anyway created by the factory.

Or move the GetHAsh/GetPowHash in to the factory itself.

Reply:
I'll create an issue to address this.